pyproject.toml

```diff
@@ -2,6 +2,67 @@
 # It's the equivalent of r-strings in Python.  Multiline strings are treated as
 # verbose regular expressions by Black.  Use [ ] to denote a significant space
 # character.
+[build-system]
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[project]
+name="aiven-client"
+authors = [
+  { name="Aiven", email="support@aiven.io" },
+]
+description="Aiven.io client library / command-line client"
+readme = "README.rst"
+requires-python = ">=3.7"
+classifiers=[
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Libraries",
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+license = { text = "Apache License 2.0" }
+dynamic = ["version"]
+dependencies = [
+    'requests>=2.2.1; sys_platform == "linux"',
+    'requests>=2.9.1; sys_platform != "linux"',
+    "certifi>=2015.11.20.1",
+]
+[project.optional-dependencies]
+dev = [
+    # Lock black to the same major version used in downstream build environments
+    "black==22.10.0",
+    "flake8",
+    # Lock mypy to the same major version used in downstream build environments
+    "mypy>=0.991,<1.0",
+    # Lock types-requests to the same major version used in downstream build environments
+    "types-requests>=2.26.1,<3.0",
+    # Lock pylint to the same major version used in downstream build environments
+    "pylint>=2.15.5,<3.0",
+    "pytest",
+    # Lock isort to the same major version used in downstream build environments
+    "isort>=5.10.1,<6.0",
+    "click",
```

Comment:
in general, locking to older versions for no good reason other than "prevent issues", is not recommended, It just moves the problems into the future.